providers/implementations/keymgmt/rsa_kmgmt.c

```diff
@@ -200,11 +200,19 @@ static int rsa_import(void *keydata, int selection, const OSSL_PARAM params[])
 
 #ifdef FIPS_MODULE
     if (ok > 0 && !ossl_fips_self_testing()) {
-        const BIGNUM *n, *e, *d;
+        const BIGNUM *n, *e, *d, *dp, *dq, *iq, *p, *q;
 
         RSA_get0_key(rsa, &n, &e, &d);
-        if (n != NULL && e != NULL && d != NULL)
-            ok = ossl_rsa_key_pairwise_test(rsa);
+        RSA_get0_crt_params(rsa, &dp, &dq, &iq);
+        p = RSA_get0_p(rsa);
+        q = RSA_get0_q(rsa);
+
+        /* Check for the public key */
+        if (n != NULL && e != NULL)
+            /* Check for private key in straightforward or CRT form */
+            if (d != NULL || (p != NULL && q != NULL && dp != NULL
+                              && dq != NULL && iq != NULL))
+                ok = ossl_rsa_key_pairwise_test(rsa);
```

Comment:
test here also?

Reply:
Not required, the called routine in rsa_gen.c does all of the self test handling.
I didn't want to move that code here.  I assume it serves a purpose being part of the core RSA implementation.